src/coreclr/jit/promotiondecomposition.cpp

```diff
@@ -731,27 +731,27 @@ class DecompositionPlan
             const Entry& entry = m_entries.BottomRef(i);
 
             assert((entry.ToLclNum != BAD_VAR_NUM) && (entry.ToReplacement != nullptr));
-            GenTree* src = m_compiler->gtNewConWithPattern(entry.Type, initPattern);
-            GenTree* dst = m_compiler->gtNewLclvNode(entry.ToLclNum, entry.Type);
-            statements->AddStatement(m_compiler->gtNewAssignNode(dst, src));
+            GenTree* value = m_compiler->gtNewConWithPattern(entry.Type, initPattern);
+            GenTree* store = m_compiler->gtNewStoreLclVarNode(entry.ToLclNum, value);
+            statements->AddStatement(store);
             entry.ToReplacement->NeedsWriteBack = true;
             entry.ToReplacement->NeedsReadBack  = false;
         }
 
         RemainderStrategy remainderStrategy = DetermineRemainderStrategy();
         if (remainderStrategy.Type == RemainderStrategy::FullBlock)
         {
-            GenTree* asg = m_compiler->gtNewAssignNode(m_dst, m_src);
-            statements->AddStatement(asg);
+            statements->AddStatement(m_store);
```

Comment:
Here and elsewhere in this function, we previously created a new node to pass to `AddStatement()`. Now, we pass `m_store` directly. Is that correct? Do we need to `gtClone(m_store)`? I don't understand the usage of `m_store`, but I want to ensure we're not linking a node in to the IR that is also used elsewhere. cc @jakobbotsch

Reply:
I do understand this to be correct, since:
1) Existing code already did this (reusing the destination/source nodes).
2) The store is added to the list (at most) once, and then all the "statements" are sequenced into a comma chain which replaces the original store in the tree.

Reply:
Yep, this looks fine to me. The original store is indeed replaced entirely by the result of the decomposition.